src/averageAges.js

```diff
@@ -15,11 +15,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  const ages = men.reduce((sum, { died, born }) => sum + (died - born), 0);
+  const menByCentury = men
+    .filter(({ died }) => Math.ceil(died / 100) === century);
+  const agesByCentury = menByCentury
+    .reduce((sum, { died, born }) => sum + (died - born), 0);
+
+  return century ? (agesByCentury / menByCentury.length) : (ages / men.length);
```

Comment:
Now everything is written correctly, but when our `century` value is false, you are doing unnecessary calculations. Try to check this condition first, and then filter if necessary. Fix this for all functions.

Also, the calculation itself looks similar to what is done next for women, so the repeated part can be taken out into a separate function and reused.